packages/react-devtools-shared/src/devtools/views/Components/KeyValue.js

```diff
@@ -195,7 +201,7 @@ export default function KeyValue({
   // Only certain types of hooks should be editable.
   let canRenameTheCurrentPath = canRenamePaths;
   if (canRenameTheCurrentPath && typeof canRenamePathsAtDepth === 'function') {
-    canRenameTheCurrentPath = canRenamePathsAtDepth(depth);
+    canRenameTheCurrentPath = !readOnly && canRenamePathsAtDepth(depth);
```

Comment:
nit: This bit isn't necessary since `canRenamePaths === false` when `readOnly` and `canRenameTheCurrentPath === canRenamePaths` by default.